### PR TITLE
Connections requests part 0

### PR DESCRIPTION
**Pull request by owensimpson**
Create backend columns for
- invitations
- sent invitations
- connection

Also updated seed script

Invitations are stored as their own Model, accessed with the new tRPC endpoint, "invitations". They connect 2 users.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -62,6 +62,19 @@ model Session {
   @@map(name: "session")
 }
 
+// Represents an Invitation to a user
+model Invitation {
```
Comment (lukejianu) [inline prisma/schema.prisma:66]: Are we sure about the naming here? I heard some discussion about calling this a ``Request`` or something.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -62,6 +62,19 @@ model Session {
...
+  id           String   @id @default(cuid()) // primary key
+  message       String
+  fromUserId String // foreign key
+  toUserId   String // foreign key
+  fromUser User  @relation("sentInvitations", fields: [fromUserId], references: [id])
+  toUser   User  @relation("receivedInvitations", fields: [toUserId], references: [id])
+
+  @@index([fromUserId])
```
Comment (lukejianu) [inline prisma/schema.prisma:74]: Can we index on ``fromUser.id`` instead of have those two extra attributes?

File: prisma/schema.prisma [hunk 3]
```diff
@@ -96,6 +109,10 @@ model User {
   endTime            DateTime? @map("end_time") @db.Time(0)
   favoritedBy        User[]    @relation("Favorites")
   favorites          User[]    @relation("Favorites")
+  sentInvitations    Invitation[]  @relation("sentInvitations")
+  receivedInvitations Invitation[] @relation("receivedInvitations")
+  connectedBy        User[] @relation("Connections")
```
Comment (lukejianu) [inline prisma/schema.prisma:114]: Should users be allowed to connect with multiple people? Can the attribute just be some ``carpoolGroup`` instead which is a list of ``User``?

File: prisma/seed.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ const prisma = new PrismaClient();
  */
 const deleteUsers = async () => {
   await prisma.user.deleteMany({});
+  await prisma.invitation.deleteMany({});
```
Comment (lukejianu) [inline prisma/seed.ts:13]: Are you sure we need this line? I know the delete behavior is weird in Prisma, but I recently noticed that if you delete a ``User`` in ``prisma studio``, it cascades and deletes the connected ``Account`` and ``Session`` entry.

File: prisma/seed.ts [hunk 5]
```diff
@@ -19,8 +20,25 @@ const clearConnections = async () => {
   const users = await prisma.user.findMany();
 
   await Promise.all(
-    users.map(async (user) => {
-      return await prisma.user.update({
+    users.map((user) =>
+      Promise.all([
+        prisma.invitation.findMany({
```
Comment (lukejianu) [inline prisma/seed.ts:25]: Why is the ``findMany`` call here?

File: prisma/schema.prisma (hunk 1 above)
Comment (devashish9) [reply prisma/schema.prisma:66]: Idc, either name is fine, would it really matter that much? 
Luke I think the discussion you heard had more to do with the frontend

File: prisma/schema.prisma (hunk 3 above)
Comment (devashish9) [reply prisma/schema.prisma:114]: Yes Luke is right here. @owensimpson connected should be just one person. I think the carpoolGroup model is a potential idea, let's talk about it more during standup

File: prisma/seed.ts (hunk 4 above)
Comment (devashish9) [reply prisma/seed.ts:13]: No harm leaving it there. Unless @owensimpson wants to look into the user delete behavior and validate the need for the extra call.

File: prisma/seed.ts (hunk 5 above)
Comment (devashish9) [reply prisma/seed.ts:25]: multiple invitations can be sent or received by a person?

File: prisma/seed.ts [hunk 6]
```diff
@@ -30,20 +48,40 @@ const clearConnections = async () => {
...
+      })
+    )
   );
 };
 
 /**
  * Generates favorites for users in our database.
  */
-const generateFavorites = async () => {
+const generateConnectedFields = async () => {
```
Comment (devashish9) [inline prisma/seed.ts:65]: nit but can we move the functionality for favorites and invites into seperate helper functions?

File: src/server/router/index.ts [hunk 7]
```diff
@@ -8,7 +9,8 @@ import { userRouter } from "./user";
 export const appRouter = createRouter()
   .transformer(superjson)
   .merge("user.", userRouter)
-  .merge("mapbox.", mapboxRouter);
+  .merge("mapbox.", mapboxRouter)
+  .merge("invitations.", invitationsRouter);
```
Comment (devashish9) [inline src/server/router/index.ts:13]: why wouldn't this just reside under user?

File: src/server/router/user.ts [hunk 8]
```diff
@@ -169,4 +169,46 @@ export const userRouter = createProtectedRouter()
         },
       });
     },
+  })
+  // Returns the list of connections for the curent user
+  .query("connections", {
```
Comment (devashish9) [inline src/server/router/user.ts:174]: Probably gonna redesign what this looks like

File: prisma/seed.ts (hunk 5 above)
Comment (lukejianu) [reply prisma/seed.ts:25]: Sure, but where is the return value getting used?

**Review by jun-w-huang — DISMISSED** [review]
amazing

**Review by DavidOB1 — DISMISSED** [review]
Marvelous!

**Review by DavidOB1 — APPROVED** [review]

**Pull request merged**